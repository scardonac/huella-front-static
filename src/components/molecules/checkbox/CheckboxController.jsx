
import { Controller } from "react-hook-form";
export default function CheckboxController({ control }) {
    return (
        // <div
        //     className="checkbox-container"
        // >
        //     <Controller
        //         name="checkbox"
        //         control={control}
        //         render={({ field }) => (
        //             <label  htmlFor={'1_1'}>
        //                 <input
        //                     id={'1_1'}
        //                     type="checkbox"
        //                     {...field}
        //                     className="checkbox"
        //                 />
        //                 Acepta el uso de datos personales y acuerdos de confidencialidad
        //                 <span className="checkmark"></span>
        //             </label>
        //         )}
        //     />
        // </div>
        // <div
        //     className={`SelectionCard bg-primary-white1 border-2 border-solid w-full px-10 flex gap-4 items-center justify-between py-2 pl-2 pr-5 rounded-xl cursor-pointer`}
        // // onClick={isActived && handleSelect}
        // >

        //     <div>
        //         <input
        //             className='checkbox'
        //             type='checkbox'
        //             id={'1_1'}
        //             checked={true}
        //             readOnly
        //         />
        //         {true && (
        //             <label className='label' htmlFor={'1_1'}>
        //                 <div className='flex ml-6'>
        //                     <span> Acepta el uso de datos personales y acuerdos de confidencialidad</span>
        //                 </div>
        //             </label>
        //         )}
        //     </div>
        // </div>
        <Controller
            name="checkboxField" // Nombre del campo en el formulario
            control={control}
            defaultValue={false} // Valor inicial del checkbox
            render={({ field }) => (
                <label className="custom-checkbox">
                    <input
                        type="checkbox"
                        className="checkbox"
                        {...field}
                    />
                    <div className="checkmark"></div>
                    Acepta el uso de datos personales y acuerdos de confidencialidad
                </label>
            )}
        />
    );
};
