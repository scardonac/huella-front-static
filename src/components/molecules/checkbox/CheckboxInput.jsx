import { useState } from 'react';

export const CheckboxInput = ({ label, name, setValue, styleDivTrue = '', styleDivFalse = null, styleLabel = '', isChecked }) => {
    const [isCheckedState, setIsCheckedState] = useState(isChecked);

    const toggleCheckbox = () => {
        setIsCheckedState(!isCheckedState);
        setValue(name, !isCheckedState);
    };

    return (
        <div
            className={isCheckedState ? styleDivTrue : (styleDivFalse || styleDivTrue)}
            id='checkbox_div'
            data-testid="checkbox_div"
        >
            <div
                className={`custom-checkbox ${isCheckedState ? 'checked' : ''}`}
                onClick={toggleCheckbox}
                id='checkbox_input'
                data-testid="checkbox_input"
            ></div>
            <label className={styleLabel} onClick={toggleCheckbox} htmlFor={name}>
                {label}
            </label>
        </div>
    );
};
