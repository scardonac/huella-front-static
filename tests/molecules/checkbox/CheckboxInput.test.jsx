import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { CheckboxInput } from "../../../src/components/molecules/checkbox/CheckboxInput";
import { act } from 'react-dom/test-utils';

describe('Pruebas en <CheckboxInput />', () => {

    test('Debe renderizarse correctamente', () => {
        const { baseElement } = render(<CheckboxInput label="Test" name="test" setValue={() => { }} isChecked={false} />);
        expect(baseElement).toMatchSnapshot();
    });

    test('Debe renderizarse correctamente con estilos', () => {
        render(<CheckboxInput label="Test" name="test" setValue={() => { }} isChecked={false} styleDivTrue="testTrue" styleDivFalse="testFalse" styleLabel="test" />);

        const fnCheckbox = screen.getByTestId('checkbox_input');
        const fnDiv = screen.getByTestId('checkbox_div');

        act(() => {
            fireEvent.click(fnCheckbox);
        });

        act(() => {
            expect(fnCheckbox.getAttribute('class')).toMatch('custom-checkbox checked');
            expect(fnDiv.getAttribute('class')).toMatch('testTrue');
        });
    });

});
