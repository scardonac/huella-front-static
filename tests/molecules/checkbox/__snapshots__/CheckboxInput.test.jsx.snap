// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CheckboxInput /> Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <div
      class=""
      data-testid="checkbox_div"
      id="checkbox_div"
    >
      <div
        class="custom-checkbox "
        data-testid="checkbox_input"
        id="checkbox_input"
      />
      <label
        class=""
        for="test"
      >
        Test
      </label>
    </div>
  </div>
</body>
`;
